crates/rpc/rpc/src/eth/api/server.rs

```diff
@@ -180,11 +181,84 @@ where
 
     async fn fee_history(
         &self,
-        _block_count: U256,
-        _newest_block: BlockNumber,
+        block_count: u64,
+        newest_block: BlockNumber,
         _reward_percentiles: Option<Vec<f64>>,
     ) -> Result<FeeHistory> {
-        Err(internal_rpc_err("unimplemented"))
+        if block_count == 0 || newest_block < block_count {
+            return Ok(FeeHistory::default())
+        }
+
+        let start_block = newest_block - block_count;
+        let end_block = newest_block;
+
+        let mut fee_history_cache = self.fee_history_cache.lock();
+
+        let mut fee_history_cache_items = BTreeMap::new();
+        let mut first_non_cached_block = None;
+        for block in start_block..=end_block {
```

Comment:
we need to loop over the block range to populate the response's vecs for
* base_fee_per_gas
* gas_used_ratio
and also calculate the rewards field

Reply:
correct, that's already done here
https://github.com/paradigmxyz/reth/blob/89932e60b1937bca8b65ffbfdb96d055d04700d4/crates/rpc/rpc/src/eth/api/server.rs#L269-L282